api/test/unit/controllers/test_search_controller.py

```diff
@@ -0,0 +1,51 @@
+from unittest import mock
+
+import pytest
+
+from catalog.api.controllers import search_controller
+from catalog.api.utils.pagination import MAX_TOTAL_PAGE_COUNT
+
+
+@pytest.mark.parametrize(
+    "total_hits, real_result_count, page_size, expected",
+    [
+        # No results
+        (0, 0, 10, (0, 0)),
+        # Setting page size to 0 raises an exception
+        pytest.param(
```

Comment:
Does it make sense to add a negative page size, too?

Reply:
Upstream validation catches negative values here.